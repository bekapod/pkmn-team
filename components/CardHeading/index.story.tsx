// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import CardHeading from ".";

storiesOf("CardHeading", module).add("default", () => (
  <CardHeading>Cupcake ipsum dolor sit amet</CardHeading>
));
