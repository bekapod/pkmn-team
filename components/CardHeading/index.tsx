import React, { ReactNode, ReactNodeArray } from "react";
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

interface Props {
  headingType?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  children: ReactNode | ReactNodeArray;
}

const CardHeading = styled(({ headingType = "h2", ...props }: Props) =>
  React.createElement(headingType, props)
)`
  overflow: hidden;
  margin: ${variables.spacing.lg}px 0;
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export default CardHeading;
