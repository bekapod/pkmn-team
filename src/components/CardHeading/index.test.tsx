import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
import CardHeading from ".";

describe("<CardHeading />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<CardHeading>Heading Default</CardHeading>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("can render a different heading type", () => {
    const tree = renderer.create(
      <CardHeading headingType="h1">Heading 1</CardHeading>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
