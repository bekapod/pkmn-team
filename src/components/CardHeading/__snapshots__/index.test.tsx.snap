// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardHeading /> can render a different heading type 1`] = `
<h1
  className="CardHeading-mmeo34-0 ebVIiR"
>
  Heading 1
</h1>
`;

exports[`<CardHeading /> renders without crashing 1`] = `
<h2
  className="CardHeading-mmeo34-0 ebVIiR"
>
  Heading Default
</h2>
`;
